Lambdas/GHHooks/Extensions/String+Document.swift

```diff
@@ -15,8 +15,17 @@ extension String {
         assert(hardLimit > 0, "Hard length limit must be greater than zero (got \(hardLimit)).")
         assert(hardLimit >= maxVisualLength, "maxVisualLength '\(maxVisualLength)' can't be more than hardLimit '\(hardLimit)'.")
 
+        /// Interpret urls like GitHub does.
+        /// For example GitHub changes `https://github.com/vapor/penny-bot/issues/99` to
+        /// `[vapor/penny-bot#99](https://github.com/vapor/penny-bot/issues/99)` which
+        /// ends up looking like a blue `vapor/penny-bot#99` text linked to the url.
+        let regex = #/https://github\.com/(?<org>[^/<>#%"{}|^\[\]\s]+)/(?<repo>[^/<>#%"{}|^\[\]\s]+)/(pull|issues)/(?<number>\d+)/#
```

Comment:
If you're gonna use the `#/ /#` "automatic extended syntax" syntax (no, that's not ungrammatical), at least take advantage of it to make your regex remotely readable. Also your two path components are too permissive; per GitHub's UI,

> The [organization] name may only contain alphanumeric characters or single hyphens, and cannot begin or end with a hyphen.

> The repository name can only contain ASCII letters, digits, and the characters ., -, and _.

Among other things. Basically, see below:

```suggestion
        let regex = #/
            https://(?:www\.)?github\.com
            /(?<org>[A-Za-z0-9](?:[A-Za-z0-9\-]*[A-Za-z0-9])?)
            /(?<repo>[A-Za-z0-9.\-_]+)
            /(?:pull|issues)
            /(?<number>\d+)
        /#
```